### ISSUE TITLE
Non ASCII no longer supported in pipeline config.

### ISSUE THREAD

**Issue by jarpy** [elastic/logstash#7851]
This seems to be [by design](https://github.com/elastic/logstash/commit/645fcec234e44c43e595710b438db757ac519f46#diff-0da0dea673120e65c9fc4e91cb5a9fb1R61), but I used to be able to do neat things like [this](https://github.com/elastic/stack-docker/blob/b02958b5b720eab4d871dfb356445046e3b14bc7/config/logstash.conf#L4).

I guess this is either an improvement or a regression, depending on one's perspective.

Relevant XKCD: https://xkcd.com/1172/

**Comment by original-brownbear** [comment]
@jarpy but I guess 💓 is UTF-8? The way I read

> The following config files contains non-ascii characters but are not **UTF-8 encoded**

this should still work. The code that determines that just changed from:

```rb
!cfg.ascii_only? && !cfg.valid_encoding?
```

```rb
content.ascii_only? && content.valid_encoding?
```

looks like this is just a typo, I'll add a quick PR for that

**Comment by original-brownbear** [comment]
Fixed in #7852 :)
